### ISSUE TITLE
Proxy settings ignored for local connections

### ISSUE THREAD

**Issue by cavaliercoder** [chocolatey/choco#497]
When connecting to https://chocolatey.org/api/v2/, the chocolatey client (v0.9.9.11) correctly uses the configured proxy server (configured via `choco config set proxy`). However, when connecting to nuget servers on my local LAN segment, the proxy settings are seemingly ignored and the client attempts to connect directly.

Often times a corporate proxy is still required to access a local web/NuGet server. Could you please consider making proxy bypass configurable?

The issue persists when using an IPv4 address, hostname or FQDN of a local NuGet host.

I've confirmed this issue in WireShark and Fiddler.

Running in PowerShell on Windows 10

```
Name                           Value
----                           -----
PSVersion                      5.0.10240.16384
WSManStackVersion              3.0
SerializationVersion           1.1.0.1
CLRVersion                     4.0.30319.42000
BuildVersion                   10.0.10240.16384
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
```

May be related to #262

**Comment by ferventcoder** [comment]
I think I might need a bit more information here. This is an explicit proxy you are setting correct?

**Comment by ferventcoder** [comment]
Yah, I think #262 is related but in the opposite way. :)

**Comment by ferventcoder** [comment]
This is strictly in getting the packages where it is is failing, correct?

**Comment by ferventcoder** [comment]
That means I'll need to make adjustments in our fork of Nuget.Core.

**Comment by ferventcoder** [comment]
If you specify an explicit proxy, we use it for getting software downloads from package instructions - https://github.com/chocolatey/choco/blob/ac0c596414d794b89834a7001a35fb1708c6774a/src/chocolatey.resources/helpers/functions/Get-WebFile.ps1#L39-L54.

However Nuget.Core (which has the responsibility of getting packages themselves) is probably trying to decide if it should bypass a proxy. Incorrectly.

**Comment by cavaliercoder** [comment]
I've tested `choco search` and `choco install`. Same result on both. End point URLs seem to be `/Search()` and `/FindPackagesById()`. Sounds like Nuget.Core to me.

I can't get much further on my home network to confirm if `Get-WebFile.ps1` has the same issue for downloading the package content. I'll confirm tomorrow from the corp network.

Thanks.

**Comment by ferventcoder** [comment]
I believe it likely doesn't have any issues because if a proxy is specified, it is not bypassed. But I agree an additional setting would be helpful

**Comment by abmoham** [comment]
Voting this up to support issue [#605](https://github.com/chocolatey/choco/issues/605)

**Comment by ferventcoder** [comment]
Thanks @abmoham. We'll bump the priority for this.

**Comment by ferventcoder** [comment]
Adding `choco config set proxyBypassList` for 0.10.4.

**Comment by ferventcoder** [comment]
Rereading this issue - moved proxyBypassList to #1165.

**Comment by ferventcoder** [comment]
I think this may actually be covered with #1165 as you will be able to specify the bypass list and whether to bypass on local or not.

**Comment by ferventcoder** [comment]
Pretty sure this is covered by #1165. Closing as fixed for 0.10.4.
